src/rust_connection/stream.rs

```diff
@@ -255,12 +261,70 @@ impl ReadFD for Stream {
             }
         }
     }
+}
 
-    fn set_nonblocking(&mut self, nonblocking: bool) -> Result<()> {
-        match self {
-            Stream::TcpStream(stream) => stream.set_nonblocking(nonblocking),
-            #[cfg(unix)]
-            Stream::UnixStream(stream) => stream.set_nonblocking(nonblocking),
+impl super::Poll for Stream {
+    fn poll(&mut self, read: bool, write: bool, block: bool) -> Result<(bool, bool)> {
+        #[cfg(unix)]
+        {
+            let mut poll_flags = nix::poll::PollFlags::empty();
+            if read {
+                poll_flags |= nix::poll::PollFlags::POLLIN;
+            }
+            if write {
+                poll_flags |= nix::poll::PollFlags::POLLOUT;
+            }
+            let fd = self.as_raw_fd();
+            let mut poll_fds = [nix::poll::PollFd::new(fd, poll_flags)];
+            let timeout = if block { -1 } else { 0 };
```

Comment:
I am only scrolling through this without looking too closely at what is going on, but... why would any code need to `poll` with a timeout of zero? In non-blocking mode, one can just do the I/O operation instead and see if it fails with `WouldBlock`.

Reply:
You're right, removed